scipy/stats/_stats_py.py

```diff
@@ -472,6 +478,13 @@ def mode(a, axis=0, nan_policy='propagate'):
         a = ma.masked_invalid(a)
         return mstats_basic.mode(a, axis)
 
+    if a.dtype.kind not in 'uifc':
```

Comment:
I'm not familiar with this style of checking. Is it preferred over
`np.issubdtype(a.dtype, np.number)`?

Reply:
I think both should be equivalent in terms of behavior. I forgot about `np.issubtype`! `np.issubdtype(a.dtype, np.number)` looks slightly more pythonic to me than `a.dtype.kind not in 'uifc'`. I'll change it.